src/qtmaximise.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef QT_MAXIMISE_H_INCLUDED
+#define QT_MAXIMISE_H_INCLUDED
+
+#include <QAction>
+#include <QMainWindow>
+
+class QActionMaximise : public QAction
+{
+    Q_OBJECT
+
+    public:
+        QActionMaximise(QMainWindow* mainWindow = 0) : QAction(tr("Maximise"),mainWindow), m_mainWindow(mainWindow)
+        {
+            setShortcut(Qt::Key_F11);
+            connect(this, SIGNAL(triggered(bool)), this, SLOT(handleTriggered(bool)));
+        }
+
+    public slots:
+        void handleTriggered(bool checked)
+        {
+            const bool maximised = m_mainWindow->windowState()&Qt::WindowMaximized;
+            if (maximised)
+                m_mainWindow->showNormal();
+            else
+                m_mainWindow->showMaximized();
+        }
```

Comment:
Yes, I think the code could do with a good reshuffle.  I opened an issue rather than discuss details here - see #74 for initial thoughts.

Regarding the `QActionMaximise` class - implementation wise I'm willing to go with it and see what it feels like.  However, I have one final objection on the design front which I only just realized was hovering in the back of my mind bugging me: window managers usually have shortcut keys for this kind of thing already - why not just learn that?  It's a win-win solution: it'll work for all your windows, and displaz will have less code.  Even windows has such a shortcut: `ctrl+win+↑`